common/djangoapps/third_party_auth/pipeline.py

```diff
@@ -397,13 +409,16 @@ def get_provider_user_states(user):
             each enabled provider.
     """
     states = []
-    found_user_backends = [
-        social_auth.provider for social_auth in models.DjangoStorage.user.get_social_auth_for_user(user)
-    ]
+    found_user_auths = list(models.DjangoStorage.user.get_social_auth_for_user(user))
 
     for enabled_provider in provider.Registry.enabled():
+        association_id = None
+        for auth in found_user_auths:
+            if enabled_provider.match_social_auth(auth):
+                association_id = auth.id
+                break
```

Comment:
If you like concise code, you could replace these five lines by

    association_id = next((auth.id for auth in found_user_auths
                           if enabled_provider.match_social_auth(auth)), None)

What you have is probably more readable, though.